datalad/distribution/remove.py

```diff
@@ -86,6 +80,21 @@ class Remove(Interface):
         if_dirty=if_dirty_opt,
     )
 
+    _examples_ = [
+        dict(text="""Permanently remove a subdataset from a dataset and wipe out
+                     the subdataset association too""",
+             code_py="remove(path='path/to/subds')",
+             code_cmd="datalad remove <path/to/subds>"),
+        dict(text="Permanently remove a dataset and all subdatasets",
+             code_py="remove('path/to/dataset/', recursive=True)",
```

Comment:
I think in other places you use `dataset=` explicitly.  It might be worth doing that consistently.

Reply:
An hour ago I thought "oh yeah, why not do for all?" and now I'm buried under notes which command resolves paths as relative to the dataset argument and which not. :smile:

I'm considering writing all of the examples as being dataset methods to avoid this confusion. Do you have thoughts on this?

Reply:
My take would be: dont worry about the old stuff. If it behaves strange, let fix it first and then document it more beautifully.

Reply:
Okay, unless someone else has a different opinion, I won't use the `dataset=` argument for remove at this point. All "core" commands either have it specified or have examples with dataset methods, and we can revisit other commands at a later point when there is more consistency in their behavior.

Reply:
> An hour ago I thought "oh yeah, why not do for all?" and now I'm buried under notes which command resolves paths as relative to the dataset argument and which doesn't.

Those notes would be valuable because we should fix these.

---

Sorry, my comment led you in a different direction than I was trying to go.  I was just wondering whether you want to use `dataset='path/to/dataset/'`, thinking 'path/to/dataset/' was being passed a positional argument for `dataset`.  Looking at it again, I was wrong in taking 'path/to/dataset/' as the `dataset` argument.  It is `path`.  But then doesn't that mean the code_py and code_cmd counterparts don't match??

```python
             code_py="remove('path/to/dataset/', recursive=True)",
             code_cmd="datalad remove -d <path/to/dataset/> --recursive"),
```

Your Python example is passing 'path/to/dataset/' as `path`, while the command-line example is passing it as `dataset`.

Ignoring that issue, my original question would become "Should you put "path='path/to/dataset/'" for consistency with, say, the example right above it?".  And perhaps in general it's worth taking the stance that our examples should use keyword arguments for clarity.

Reply:
> Okay, unless someone else has a different opinion, I won't use the `dataset=` argument for remove at this point. All "core" commands either have it specified or have examples with dataset methods, and we can revisit other commands at a later point when there is more consistency in their behavior.

At least for the examples you're adding in this PR, I'd recommend consistently using unbound command calls for the Python examples, specifying the `dataset=` argument (as a string) when the command-line variant has  `-d/--dataset` specified.  Even for commands that don't follow the new path semantics, this should get you matching behavior between the command-line and Python variants.

Reply:
Thanks a lot for clarifying, I misunderstood. I agree to using keyword arguments for clarity and matching arguments in Python and CMD examples. I'll push an update to the examples in this PR.

As a more general note to think about, some examples use dataset methods (e.g., run and save - if I recall correctly because these were preexisting examples in the docstrings), and it may be useful to unify whether examples show functions or methods at some point.

> Those notes would be valuable because we should fix these.

Agreed, I will go the remaining commands and file and issue with my observations.